data-prepper-plugins/sqs-source/src/main/java/org/opensearch/dataprepper/plugins/source/sqssource/SqsSource.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source.sqssource;
+
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSetManager;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.plugin.PluginFactory;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.Source;
+import org.opensearch.dataprepper.plugins.aws.sqs.common.BufferAccumulator;
+import org.opensearch.dataprepper.plugins.aws.sqs.common.codec.Codec;
+import org.opensearch.dataprepper.plugins.aws.sqs.common.codec.JsonCodec;
+import org.opensearch.dataprepper.plugins.aws.sqs.common.metrics.SqsMetrics;
+
+@DataPrepperPlugin(name = "sqs", pluginType = Source.class,pluginConfigurationType = SqsSourceConfig.class)
+public class SqsSource implements Source<Record<Event>> {
+
+    private final SqsSourceConfig sqsSourceConfig;
+
+    private final AcknowledgementSetManager acknowledgementSetManager;
+
+    private final PluginMetrics pluginMetrics;
+
+    private SqsSourceService sqsSourceService;
+
+    private final boolean acknowledgementsEnabled;
+
+    private final Codec codec;
+
+    @DataPrepperPluginConstructor
+    public SqsSource(final PluginMetrics pluginMetrics,
```

Comment:
This source should use the new AWS Plugin. See this PR for an example: https://github.com/opensearch-project/data-prepper/pull/2787

Reply:
Thanks @dlvenable for your review, Sure i will make the changes accordingly.